Talk about documentation

jaylett opened the pull request:
* what there is
 * how to build it (briefly)
 * some brief text about writing documentation to support the code
   you work on
 * how to build the PDF version of the documentation

jaylett committed:
Talk about documentation

 * what there is
 * how to build it (briefly)
 * some brief text about writing documentation to support the code
   you work on
 * how to build the PDF version of the documentation

ojwb commented on documentation/index.rst:
```diff
@@ -0,0 +1,158 @@
+   typically either as a "full worked example", or as a deep dive into
+   a particular feature.
+
+   Most of this documentation lives in the*Xapian user guide*, which
```
Looks like a missing space ("the*).

ojwb commented on documentation/index.rst:
```diff
@@ -0,0 +1,158 @@
+4. `The Xapian wiki`_ also contains information, including links
+   to articles and talks that people have written about using Xapian
+   in various ways. Although this isn't always official Xapian
+   documentation, it may be helpfuil. Some of it may now be quite out
```
Typo: "helpfuil"

ojwb commented on documentation/index.rst:
```diff
@@ -0,0 +1,158 @@
+
+.. code-block:: bash
+
+   make html LANGUAGE=c++
```
I wonder if we want to start duplicating documentation about these sort of details?  If we just said something like "see `make help` for details of the options available" that would avoid having an extra place to update if we change the details of how the language is selected here.

jaylett replied:
I've removed the duplication; in the process I noticed that neither user nor dev guides had amazing documentation. I've fixed the dev guide in another commit in this branch, and I'll open a separate PR for the user guide that makes the two more consistent.

jaylett committed:
fixup! Talk about documentation

jaylett committed:
Be explicit about how to build the docs

jaylett committed:
Update to Sphinx 1.8.x

This seems to work without problems (and it should do, looking at
the upgrade notes), and it's much better to be explicitly using a
recent version.

ojwb commented:
Looks good.  Rebased and autosquashed and pushed.  I tweaked `requirements.txt` to say `Sphinx~=1.8.0` instead of `==` to align it with the user guide.
